syft/generic/object_storage.py

```diff
@@ -70,15 +70,21 @@ def get_obj(self, obj_id: Union[str, int]) -> object:
         Returns:
             Object with id equals to `obj_id`.
         """
-
+        obj = None
         try:
-            obj = self._objects[obj_id]
-        except KeyError as e:
-            if obj_id not in self._objects:
-                raise ObjectNotFoundError(obj_id, self)
-            else:
-                raise e
-
+            # if object is a tesor
+            obj = self._objects.get(obj_id, None)
+            if obj is None:
+                # check for placeholder_ids
+                found = (self._objects[x] for x in self._objects if x.value == obj_id)
```

Comment:
Maybe the placeholders should be stored in `_objects` with id.value as a key? Would it make things easier?

Reply:
If it would help, we could split `_objects` by type or something. 🤔